lib/devise/multi_email/models/confirmable.rb

```diff
@@ -38,64 +38,108 @@ module ConfirmableExtensions
 
         included do
           multi_email_association.include_module(EmailConfirmable)
-        end
-
-        # delegate before creating overriding methods
-        delegate :skip_confirmation!, :skip_confirmation_notification!, :skip_reconfirmation!, :confirmation_required?,
-                 :confirmation_token, :confirmed_at, :confirmation_sent_at, :confirm, :confirmed?, :unconfirmed_email,
-                 :reconfirmation_required?, :pending_reconfirmation?, to: :primary_email_record, allow_nil: true
-
-        # This need to be forwarded to the email that the user logged in with
-        def active_for_authentication?
-          login_email = current_login_email_record
 
-          if login_email && !login_email.primary?
-            super && login_email.active_for_authentication?
-          else
-            super
+          # Handle automatically confirming and linking `unconfirmed_email` if email change is not postponed
+          before_update do
```

Comment:
@allenwq in the link you posted, the action filter will postpone changing the email (by which it saves the new email in the `unconfirmed_email` column) *if* the `postpone_email_change?` method returns `true`. If it *does not* return true, then the action filter is never run. What that means is that rather than copying the changed email address from `email` to `unconfirmed_email`, it leaves the change in `email` and saves that value without confirming the new email address.

I the refactor, there is no modifying of existing `Email` records. When the `email` value is changed, we either match an existing `Email` record or create a new one for the specified email. What that means is, if the `email` value is changed and the change is not copied to the `unconfirmed_email` column in Devise, we need to automatically mark the new email address as confirmed (because Devise didn't confirm the change either) in this specific scenario.

(I know this may be a bit confusing, but just think about how Devise's logic relies on the "dirty" `email` value and how that behavior is different when that is actually a separate record instead of a database column. We are just mimicking the same result.